tests/integration/istioio/configuration/scripts/liveness_and_readiness_probes_with_http_globally.txt

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+# $snippet update_config_map.sh syntax="bash"
+kubectl -n istio-io-health get cm istio-sidecar-injector -n istio-system -o yaml | sed -e 's/"rewriteAppHTTPProbe":false/"rewriteAppHTTPProbe":true/' | kubectl apply -f -
+# $endsnippet
+
+# $snippet globally_re-deploy.sh syntax="bash"
+$ kubectl -n istio-io-health apply -f <(istioctl kube-inject -f @samples/health-check/liveness-http-same-port.yaml@)
+# $endsnippet
+
+$ kubectl -n istio-io-health rollout status deployment liveness-http --timeout 60s
+
+# $snippet re-deploy_globally_check_status.sh syntax="bash" outputis="text"
+$ kubectl -n istio-io-health get pod
+# $verify verifier="lineRegex"
+NAME *READY *STATUS *RESTARTS *AGE
+liveness-http-[0-9a-z]*-[0-9a-z]* *2/2 *Running *0 ?
+# $endsnippet
+
+# $snippet delete_globally_deploy.sh syntax="bash"
+$ kubectl -n istio-io-health delete -f <(istioctl kube-inject -f @samples/health-check/liveness-http-same-port.yaml@)
+# $endsnippet
+
+# let delete operation finish
+sleep 30s
```

Comment:
I think we shouldn't have to sleep in these tests, they take long enough as it is. Maybe you should consider deploying to different namespaces if you run into timing issues?

Reply:
I also don't like to add more and more `sleeps`. One of the issue that i faced is previous liveness pod is still in terminating state when new liveness-http is being created, because of which test started to fail as it's not able to match expected output.
```
--- FAIL: TestHealthCheck (55.16s)
    verifier.go:117: verification failed for command re-deploy_globally_check_status.sh: line count: (expected 2, found 3). Expected:
        NAME                             READY   STATUS        RESTARTS   AGE
        liveness-6dbbbdc676-xmg6j        2/2     Terminating   0          23s
        liveness-http-57c9bfff5f-pbg6g   2/2     Running       0          8s
        
        to match:
        NAME *READY *STATUS *RESTARTS *AGE
        liveness-http-[0-9a-z]*-[0-9a-z]* *2/2 *Running *0 ?
FAIL
FAIL    istio.io/istio/tests/integration/istioio/configuration  210.418s
```
I will try deploying to different namespaces and see how it works.

Reply:
Running each test in different namespace and it's working fine. thanks for the suggestion.